File: Lesson4JavaCode/sigpwned-stork-0a9c0d4/src/main/java/com/sigpwned/stork/engine/runtime/expr/IntExpr.java
package com.sigpwned.stork.engine.runtime.expr;

import com.sigpwned.stork.engine.runtime.Expr;
import com.sigpwned.stork.engine.runtime.Scope;

public class IntExpr extends Expr {
	private long value;
	
	public IntExpr(long value) {
		this.value = value;
	}
	
	public long getValue() {
		return value;
	}

	public Object eval(Scope scope) {
		return Long.valueOf(getValue());
	}
}

File: Stork.Core/Engine/Runtime/Expr/IntExpr.cs
namespace Stork.NET.Core.Engine.Runtime
{
    public sealed class IntExpr : Expr
    {
        private readonly long _value;
        public long Value
        {
            get { return _value; }
        }

        public IntExpr(long value)
        {
            _value = value;
        }
    }
}
